docs/DeveloperGuide.md

```diff
@@ -660,3 +667,114 @@ testers are expected to do more *exploratory* testing.
+            throw new CommandException(MESSAGE_NOT_EDITED);
+        }
+        List<JobApplication> lastShownList = model.getDisplayedApplicationList();
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_APPLICATION_DISPLAYED_INDEX);
+        }
+
+        JobApplication jobApplication = lastShownList.get(targetIndex.getZeroBased());
+        JobApplication newApplication = createApplication(jobApplication, editApplicationDescriptor);
+
+        try {
+            model.replaceApplication(targetIndex, newApplication);
+        } catch (IllegalValueException e) {
+            throw new CommandException(e.getMessage());
+        }
+
+        return new CommandResult(String.format(MESSAGE_EDIT_APPLICATION_SUCCESS, newApplication));
+    }
+```
+
+We can easily add a new line to check if the all the contents of the new application is the same as the old one, and throw a CommandException if it is.
+
+This can also be easily done for editing contacts.
+
+### Make commands only take in arguments that are applicable to them and reject other extra arguments.
+
+Currently, some commands with multiple modes share parameters between modes.
+
+Example: The edit command has a contact mode and job application mode. In contact mode, `--name` is used to edit the name of the contact. However, there is no such thing for job application mode.
+
+Yet, `edit --application 1 --name John --title SWE` works as long as a valid field that is used for job application is used, and ignores the extra arguments.
+
+This can be done with a simple fix. Every command alreadly has a list of flags that are accepted. 
+
+At the command parsing level, add additional checks against the list of flags provided by the command to ensure that every flag present in the command is applicable to the command used.
+
+<img src="images/enhancements/FlagChecker.png">
+
+Note: Due to the limitations of PlantUML, `--application 1 --name Jay --title SWE` is interpreted as: ~~application 1~~ name Jay --title SWE
+
+### Better Formatting for Contacts
+
+Currently, the contacts are not nicely formatted and exposes some internal but non-critical implementation details.
+
+<img src="images/enhancements/formatting.png">
+
+This is due to there not being a proper string conversion for the fields in the Contact class, especially when the fields which used to be compulsory are now optional.
+
+In this case, it would be easy to address this problem, by using 
+```java
+optionalField.map(OptionalFieldClass::toString).orElse("None")
+```
+
+As for the class name, the Contact class can use a protected abstract `getClassName` method and have the Organization and Recruiter classes extending it to implement it.
+
+Then, the Contact class can use the `getClassName` method to get the type of the contact. Alternatively, it can use the `getType` method and use it for the class name, since the `getType` method matches the class name.
```

Comment:
Java already provides access to formatted class names directly. We just need to use the right method call that doesn't expose the entire package - an interface method is not strictly necessary. However, this can be beneficial if for some reason our internal class names are different from the actual user-exposed names.